Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1104,31 +1116,41 @@ private extension PurchasesOrchestrator {
         let unsyncedAttributes = self.unsyncedAttributes
         let adServicesToken = self.attribution.unsyncedAdServicesToken
 
-        self.transactionPoster.handlePurchasedTransaction(
-            purchasedTransaction,
-            data: .init(
-                appUserID: self.appUserID,
-                presentedOfferingID: offeringID,
-                unsyncedAttributes: unsyncedAttributes,
-                aadAttributionToken: adServicesToken,
-                storefront: storefront,
-                source: self.purchaseSource(for: purchasedTransaction.productIdentifier,
-                                            transaction: purchasedTransaction,
-                                            restored: restored)
-            )
-        ) { result in
-            self.handlePostReceiptResult(result,
-                                         subscriberAttributes: unsyncedAttributes,
-                                         adServicesToken: adServicesToken)
+        self.operationDispatcher.dispatchOnWorkerThread {
```

Comment:
⚠️ Another change here. We were handling transactions in the main thread without this, which now made us read/write cache in the main thread.
Thanks to the new assertions we're avoiding this now.